### ISSUE TITLE
Web ui returning 301 redirects or 304 instead of the dashboard

### ISSUE THREAD

**Issue by meskarune** [hashicorp/consul#2094]
**Server**: Consul v0.6.4
**Clients**: Consul v0.6.4

### `consul info` for both Client and Server
Client Callisto:
```
agent:
        check_monitors = 1
        check_ttls = 0
        checks = 1
        services = 3
build:
        prerelease = 
        revision = 
        version = 0.6.4
consul:
        known_servers = 1
        server = false
runtime:
        arch = amd64
        cpu_count = 2
        goroutines = 34
        max_procs = 2
        os = freebsd
        version = go1.6.2
serf_lan:
        encrypted = true
        event_queue = 0
        event_time = 20
        failed = 0
        intent_queue = 0
        left = 0
        member_time = 92
        members = 3
        query_queue = 0
        query_time = 1
```
Client thebe:
```
agent:
        check_monitors = 0
        check_ttls = 0
        checks = 0
        services = 0
build:
        prerelease = 
        revision = 
        version = 0.6.4
consul:
        known_servers = 1
        server = false
runtime:
        arch = amd64
        cpu_count = 1
        goroutines = 33
        max_procs = 1
        os = freebsd
        version = go1.6.1
serf_lan:
        encrypted = true
        event_queue = 0
        event_time = 20
        failed = 0
        intent_queue = 0
        left = 0
        member_time = 93
        members = 3
        query_queue = 0
        query_time = 1
```
Server Juno:
```
agent:
        check_monitors = 0
        check_ttls = 0
        checks = 0
        services = 1
build:
        prerelease = 
        revision = 
        version = 0.6.4
consul:
        bootstrap = true
        known_datacenters = 1
        leader = true
        server = true
raft:
        applied_index = 7895
        commit_index = 7895
        fsm_pending = 0
        last_contact = never
        last_log_index = 7895
        last_log_term = 5771
        last_snapshot_index = 0
        last_snapshot_term = 0
        num_peers = 0
        state = Leader
        term = 5771
runtime:
        arch = amd64
        cpu_count = 1
        goroutines = 63
        max_procs = 1
        os = freebsd
        version = go1.6.1
serf_lan:
        encrypted = true
        event_queue = 0
        event_time = 20
        failed = 0
        intent_queue = 0
        left = 0
        member_time = 92
        members = 3
        query_queue = 0
        query_time = 1
serf_wan:
        encrypted = true
        event_queue = 0
        event_time = 1
        failed = 0
        intent_queue = 0
        left = 0
        member_time = 6
        members = 1
        query_queue = 0
        query_time = 1
```
### Operating system and Environment details

Freebsd 10.3 - 64 bit inside a vmware container on all hosts

There is 1 server and 2 clients. The clients have the same configuration. Consul is running over public IP's as my hosting provider doesn't give private IP's to their VPS's. I have firewall rules setup so only the consul cluster members can access consul ports.

On the server I have lighttpd installed and running with mod_proxy. I want to proxy localhost:8500 to http://stats.doloresportalatin.info

### Description of the Issue (and unexpected/desired result)

When I try to access the web ui I am getting 301 and 304 errors.

### Reproduction steps

Server Consul configuration:
```
{
    "bootstrap": true,
    "server": true,
    "datacenter": "cloud_at_cost",
    "node_name": "juno",
    "data_dir": "/var/tmp/consul",
    "encrypt": "<KEY>",
    "bind_addr": "104.233.122.181",    
    "addresses": {
        "http": "127.0.0.1"
    },
    "ports": {
        "http": 8500
    },
    "log_level": "INFO",
    "enable_syslog": false,
    "retry_join": ["104.233.122.181", "104.233.125.240", "64.137.180.62"]
}
```
Client Consul configurations:
```
{
    "server": false,
    "datacenter": "cloud_at_cost",
    "node_name": "callisto",
    "data_dir": "/var/tmp/consul",
    "encrypt": "<KEY>",
    "bind_addr": "104.233.125.240",
    "log_level": "INFO",
    "enable_syslog": false,
    "retry_join": ["104.233.122.181", "104.233.125.240", "64.137.180.62"]
}
```
```
{
    "server": false,
    "datacenter": "cloud_at_cost",
    "node_name": "thebe",
    "data_dir": "/var/tmp/consul",
    "encrypt": "<KEY>",
    "bind_addr": "64.137.180.62",
    "log_level": "INFO",
    "enable_syslog": false,
    "retry_join": ["104.233.122.181", "104.233.125.240", "64.137.180.62"]
}
```

Lighttpd vhost:

```
$HTTP["host"] == "stats.doloresportalatin.info" {
  server.document-root = vhosts_dir + "/stats.doloresportalatin.info/public"
  alias.url += ( "/munin-static" => "/usr/local/etc/munin/static" )
  alias.url += ("/munin" => "/usr/local/www/munin")
  $HTTP["url"] =~ "^/ui/" {
    proxy.server = (""    => (( "host" => "127.0.0.1",
                                "port" => 8500))
    )
  }
}
```
I have munin running on the same server and it works fine: http://stats.doloresportalatin.info/munin
I would prefer to have consul at `stats.doloresportalatin.info/consul` but first I want to get the proxy to actually work. Then I plan on requiring auth to access it over the public internet.

### Logs

I tried to access http://doloresportalatin.info/ui/ and got the following logs from lighttpd with debugging turned on:
[lighttpd debugging logs](https://gist.github.com/meskarune/c35b9935fd1d1603ed42956d5cb9e1bd)
The tldr is lighttpd is getting a 304 from consul, and in the browser I see consul's 404 page.

I used curl to test the web ui on localhost on the consul server
[testing localhost:8500](https://gist.github.com/meskarune/845e72b71a0223e95dd6d62752e70cad)

tldr I am getting this:
```
< HTTP/1.1 301 Moved Permanently
< Location: /ui/
```

**Comment by meskarune** [comment]
I got consul working on it's own virtual host, http://juno.doloresportalatin.info. I was unable to get it to work under a web directory.

**Comment by slackpad** [comment]
Hi @meskarune did you figure this one out? I'm not sure how to configure lighthttpd, bit it looks like it's trying fetch "/" from Consul instead of "/ui/". That redirect comes from here - https://github.com/hashicorp/consul/blob/master/command/agent/http.go#L419-L436. If you can get lighthttpd to pass on the full URL that should fix it. Hope that helps!
